run/types/DeviceWrapper.ts

```diff
@@ -1551,60 +1574,74 @@ export class DeviceWrapper implements SharedDeviceInterface {
+    await sleepFor(100);
+    await this.clickOnByAccessibilityID("User settings");
+    await sleepFor(500);
+    await this.clickOnElementAll(new PrivacyButton(this));
+    await sleepFor(2000);
+    await this.clickOnElementAll(new ReadReceiptsButton(this));
+    await this.navigateBack(platform);
+    await sleepFor(100);
+    await this.clickOnElementAll(new ExitUserProfile(this));
   }
 
   public async checkPermissions(platform: SupportedPlatformsType) {
-    let permissions;
-    if (platform === "ios") {
-      permissions = await this.doesElementExist({
-        strategy: "accessibility id",
-        selector: "Don’t Allow",
-        maxWait: 1000,
-      });
-    } else {
-      permissions = await this.doesElementExist({
+    if (platform === "android") {
+      const permissions = await this.doesElementExist({
         strategy: "id",
         selector: "com.android.permissioncontroller:id/permission_deny_button",
         maxWait: 1000,
       });
+      if (permissions) {
+        this.clickOnElementAll({
+          strategy: "id",
+          selector:
+            "com.android.permissioncontroller:id/permission_deny_button",
+        });
+      }
+      return;
     }
-    if (platform === "ios" && permissions) {
-      this.clickOnByAccessibilityID("Don’t Allow");
-    } else if (platform === "android" && permissions) {
-      this.clickOnElementAll({
-        strategy: "id",
-        selector: "com.android.permissioncontroller:id/permission_deny_button",
+    if (platform === "ios") {
+      // Retrieve the currently active app information
+      const activeAppInfo = await this.execute("mobile: activeAppInfo");
+      // Switch the active context to the iOS home screen
+      await this.updateSettings({
+        defaultActiveApplication: "com.apple.springboard",
       });
+
+      try {
+        // Execute the action in the home screen context
+        const iosPermissions = await this.doesElementExist({
+          strategy: "accessibility id",
+          selector: "Don’t Allow",
+        });
+        if (iosPermissions) {
+          await this.clickOnByAccessibilityID("Don’t Allow");
+        }
+      } catch (e) {
+        console.warn("iosPermissions doesElementExist failed with: ", e);
+        // Ignore any exceptions during the action
+      }
+
+      // Revert to the original app context
+      await this.updateSettings({
+        defaultActiveApplication: activeAppInfo.bundleId,
+      });
+      return;
     }
   }
+  // eslint-disable-next-line @typescript-eslint/require-await
+  public async execute(toExecute: string) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return (this.device as any).execute(toExecute);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  public async updateSettings(details: Record<string, any>) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return (this.device as any).updateSettings(details);
```

Comment:
if you still can't get to remove those any, come grab me so we have a look

Reply:
Can't remove